site/_posts/2022-11-01-authenticators-ceremonies-webauthn-oh-my.md

```diff
@@ -0,0 +1,128 @@
+
+Here are a few of the reasons WebAuthn is the superior choice for keeping your application secure:
+
+* WebAuthn helps resolve significant security problems related to data breaches, phishing, and attacks against 2FA methods. Phishing attacks are usually disguised emails or messages luring users to click on malicious links with the intent of stealing sensitive data like login credentials. With WebAuthn, attackers can't gain access to your data using login credentials stolen from your device since WebAuthn uses cryptographic keys to authenticate users and the private key never leaves the device nor is accessible to the user.
+* It's easier to use since users have to manage fewer passwords.
+* It eliminates the need for servers to store passwords. With passwords, if a server happens to be compromised, the user might be at risk if the password wasn’t hashed properly. With WebAuthn, if the server is compromised, the server only stores the public key, which requires the private key (stored in the user's device) to complete the authentication.
+* It's easier for attackers to steal a user's password than to steal their device. Even if an attacker does steal a device, they still need to bypass the device's security (*ie* biometrics or facial recognition system).
+* WebAuthn is easy to implement because it consists of [browser APIs that can be easily used by developers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API#examples).
+* It offers a smooth authentication experience for users.
+
+## How WebAuthn Works
+
+WebAuthn consists of three components: the authenticator (the user's device), the web browser, and the server (RP). In essence, the user initiates login via the browser, the browser then prompts the user to authenticate using their device (via biometrics or facial recognition, for example), and then if successful, the user is logged into the web application.
+
+However, this is only a basic view of the process. To understand the process in more depth, you need to look at the two main flows of WebAuthn: the registration flow and the authentication flow. (They are called "ceremonies" in the specification.)
+
+### The Registration Flow
+
+Before users can authenticate, their device(s) must be registered on the server of the RP. This way, the RP creates the credentials that the user can authenticate, and the process looks like this:
+
+{% include _image.liquid src="/assets/img/blogs/webauthn-terminology/registration-flow.png" alt="WebAuth Registration Flow diagram" class="img-fluid" figure=true %}
+
+1. The user visits the web application of the RP and initiates registration.
+2. The server creates a challenge key for registration, which is only used once.
+3. The challenge key, along with the user's info, is sent to the user's device.
+4. The user is prompted to authenticate themselves (this could be via biometrics, facial recognition system, or other forms depending on the device).
```

Comment:
```suggestion
4. The user is prompted to authenticate themselves (this could be via fingerprint, facial recognition, or other forms depending on the device).
```